api/sessions.js

```diff
@@ -46,14 +47,45 @@ exports.create = function(request, response) {
 exports.destroy = function(request, response) {
     request.session && request.session.remove(function(err, doc) {
         if (err) {
-            error(response, 500, 
-                'There was a problem logging you out - please retry.');
+            error(response, 500, 'There was a problem logging you out - please retry.');
         } else {
             ok(response);
         }
     });
 };
 
+// Public webhook for Authy to POST to
+exports.authyCallback = function(request, response) {
+    var authyId = request.body.authy_id;
+
+    // Look for a user with the authy_id supplies
+    User.findOne({
+        authyId: authyId
+    }, function(err, user) {
+        if (err || !user) return invalid();
+        user.authyStatus = request.body.status;
+        user.save();
+    });
+    response.end();
+};
+
+// Internal endpoint for checking the status of OneTouch
+exports.authyStatus = function(request, response) {
+    var status = request.user.authyStatus;
```

Comment:
should do a quick null check - `var status = (request.user) ? request.user.authyStatus : 'some default no status';`